broker/src/main/java/io/camunda/zeebe/broker/transport/AsyncApiRequestHandler.java

```diff
@@ -79,6 +90,10 @@ private void handleRequest(
       final DirectBuffer buffer,
       final int offset,
       final int length) {
+    final var requestReader = requestReaderSupplier.get();
+    final var responseWriter = responseWriterSupplier.get();
+    final var errorResponseWriter = errorResponseWriterSupplier.get();
```

Comment:
:thought_balloon: I still think you don't have to create these instances here. Instead it can be created in the implementation of `handleAsync`. But this is ok for now. We can discuss it when you refactor this class.